tests/components/lupusec/test_config_flow.py

```diff
@@ -0,0 +1,94 @@
+""""Unit tests for the Lupusec config flow."""
+
+import pytest
+from voluptuous.error import MultipleInvalid
+
+from homeassistant import config_entries
+from homeassistant.components.lupusec.config_flow import (
+    is_valid_host,
+    validate_configuration,
+)
+from homeassistant.components.lupusec.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+
+async def test_form_empty_input(hass: HomeAssistant) -> None:
+    """Test handling empty user input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    try:
+        await hass.config_entries.flow.async_configure(result["flow_id"], {})
+        pytest.fail("Expected error not raised")
+    except MultipleInvalid as e:
+        # Check if the error contains the expected missing key
+        assert any(
+            "required key not provided @ data['username']" in str(err)
+            for err in e.errors
+        )
+
+
+async def test_form_invalid_host(hass: HomeAssistant) -> None:
+    """Test handling invalid host input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            "host": "invalid_host",
+            "username": "test-username",
+            "password": "test-password",
+        },
+    )
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"host": "invalid_host"}
+
+
+async def test_form_valid_input(hass: HomeAssistant) -> None:
+    """Test handling valid user input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
```

Comment:
I think these constants are deprecated

```suggestion
    assert result["type"] == FlowResultType.FORM
```

Reply:
Fixed.